src/Tizen.NUI.Components/Controls/Navigation/ContentPage.cs

```diff
@@ -136,34 +135,41 @@ public View Content
                     return;
                 }
 
-                content.Weight = 1.0f;
+                Add(content);
 
-                ResetContent();
+                CalculatePosition();
             }
         }
 
-        private void ResetContent()
+        /// <inheritdoc/>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override void OnRelayout(Vector2 size, RelayoutContainer container)
         {
-            // To keep the order of AppBar and Content, the existing contents are
-            // removed and added again.
-            if ((appBar != null) && Children.Contains(appBar))
-            {
-                Remove(appBar);
-            }
+            base.OnRelayout(size, container);
 
-            if ((content != null) && Children.Contains(content))
-            {
-                Remove(content);
-            }
+            CalculatePosition();
+        }
 
-            if (appBar != null)
+        // Calculate appBar and content's positions.
+        private void CalculatePosition()
+        {
+            if (appBar == null)
             {
-                Add(appBar);
+                if (content != null)
+                {
+                    content.Position2D = new Position2D(Position2D.X, Position2D.Y);
+                    content.Size2D = new Size2D(Size2D.Width, Size2D.Height);
```

Comment:
It may be updated later to apply padding and margins.
content.Position2D = new Position2D(Position2D.X + Padding.Start + content.Margin.Start, Position2D.Y + Padding.Top + content.Margin.Top);
content.Size2D = new Size2D(Size2D.X - Padding.Start - Padding.End - content.Margin.Start - content.Margin.End , Position2D.Y - Padding.Top  - Padding.Bottom - content.Margin.Top - content.Margin.Bottom);

also I wonder it is okay to change content size even though app-developer demand another Size value.
shouldn't it be restricted only for LayoutParam.MatchedParent?

Reply:
Thank you for your comment!
The code has been updated.